standalone-metastore/metastore-catalog/src/main/java/org/apache/iceberg/rest/HMSCatalogServlet.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.UncheckedIOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.hadoop.hive.metastore.SecureServletCaller;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.HttpHeaders;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.io.CharStreams;
+import org.apache.iceberg.rest.HMSCatalogAdapter.HTTPMethod;
+import org.apache.iceberg.rest.HMSCatalogAdapter.Route;
+import org.apache.iceberg.rest.responses.ErrorResponse;
+import org.apache.iceberg.util.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Original @ https://github.com/apache/iceberg/blob/main/core/src/test/java/org/apache/iceberg/rest/RESTCatalogServlet.java
+ * The RESTCatalogServlet provides a servlet implementation used in combination with a
+ * RESTCatalogAdaptor to proxy the REST Spec to any Catalog implementation.
+ */
+public class HMSCatalogServlet extends HttpServlet {
```

Comment:
I would recommend to use Spring as it is more modern than Servlet.
Check this:
https://www.youtube.com/watch?v=tG7WigPDfqQ&ab_channel=ProgrammingwithBaky

Reply:
I have seen a thread couple weeks ago where it said, the spring 6 is not free anymore and it has a high cost.  But for sure servlet is an old school.

Reply:
Adding a new (big) dependency and a new skillset imho. Simple solution for simple feature. :-)

Reply:
> I have seen a thread couple weeks ago where it said, the spring 6 is not free anymore and it has a high cost. But for sure servlet is an old school.

Old school does not equate to bad solution though. ;-)

Reply:
@aturoczy I tried to find any news about Spring not free, but TBH I could not find any page mentioning this.

Reply:
@henrib what do you mean by "simple" feature? There exist other complete projects/apps/"features" as Iceberg REST Server like Apache Gravitino, Polaris, Unity, Project Nessie, AWS Glue, etc.  This PR has 22 file changes now and 9 reviewers on it and it is the 2nd edition.

Reply:
I had time to rethink my opinion about Spring (preferably Spring Boot): 
The problem is Java 8: 
The latest Spring Boot version that supports Java 8 is 2.7.x. 2.7.18 was a release in 2023 and they announced that this is the end of the 2.7.x line. After that, there is only commercial support. So, if we use legacy technology like Java 8, we can say it is not free. 
I did some search about alternatives: Quarkus announced dropping java 8 in 2020: https://github.com/quarkusio/quarkus/wiki/Why-Dropping-Java-8. 
I saw some other ones. Most of them has some legacy version for Java 8. 
At the end, it seems I cannot recommend a better approach than using WebServlets. 

That choice looks too traditional. But as we use Java 8, I don't know a better alternative.

Reply:
@zratkai I just meant using a "simple" Servlet directly vs using any richer superset (ie Spring) for ~15 endpoints whilst reusing as much as possible existing code. I'm certainly not considering Iceberg Catalog as a simple feature as a whole...

Reply:
We could use [armeria](https://armeria.dev/) [Java 8 (or later)]. UnityCatalog uses it
https://github.com/unitycatalog/unitycatalog/blob/main/server/src/main/java/io/unitycatalog/server/service/IcebergRestCatalogService.java